### PR TITLE
change return type for element finding

### PR DESCRIPTION
**Pull request by Yinxl**

### CODE REVIEW

File: src/main/java/macaca/client/commands/Element.java [hunk 1]
```diff
@@ -45,12 +45,30 @@ public boolean hasElement(JSONObject jsonObject) throws Exception {
...
+	 *
+	 * @throws Exception
+	 */
+	public boolean findElement(JSONObject jsonObject) throws Exception {
 		jsonObject.put("sessionId", driver.getSessionId());
 		JSONObject response = (JSONObject) utils.request("POST", DriverCommand.FIND_ELEMENT, jsonObject);
 		JSONObject element = (JSONObject) response.get("value");
 		Object elementId = (Object) element.get("ELEMENT");
-		driver.setElementId(elementId);
+		if (elementId !=  null) {
```
Comment (xudafeng) [inline src/main/java/macaca/client/commands/Element.java:63]: 改成这样的原因是什么？

**Comment by Yinxl** [comment]
# 1.0.48 /2016-12-28

*  Change return type for all elementBy..  APIs

	before this change,all elementBy.. APIs returns the current instance of MacacaClient,that means we can not operate an element directly ,it limits our usage about elements, in this change,we change the return type for all elementBy.. method to an Element object,so we can operate our target element more flexible.

	take one of these APIs as example:

	before:
	
		// MacacaClient.java
		/**
		 * <p>
	 	* Search for an element on the page, starting from the document root.<br>
	 	* Support: Android iOS Web(WebView)
	 	*
		 * @param elementId
		 *            The ID attribute of element
	 	* @return return the current instance of MacacaClient
	 	* @throws Exception
	 	*/
		public Element elementById(String elementId) throws Exception {
			JSONObject jsonObject = new JSONObject();
			jsonObject.put("value", elementId);
			jsonObject.put("using", "id");
			element.findElement(jsonObject);
			return this;
		}

	now:
	
		// MacacaClient.java
		/**
	 	* <p>
	 	* Search for an element on the page, starting from the document root.<br>
	 	* Support: Android iOS Web(WebView)
	 	*
		 * @param elementId
	 	*            The ID attribute of element
	 	* @return return the element to find if it exists,if it does not exist ,return null
		 * @throws Exception
		 */
		public Element elementById(String elementId) throws Exception {
			JSONObject jsonObject = new JSONObject();
			jsonObject.put("value", elementId);
			jsonObject.put("using", "id");
			boolean isExist = element.findElement(jsonObject);
			return isExist ? element : null;
		}
	
	

	with this change,you can operate your element as you need, like this:
	
		Element element = driver.elementById("your_element_id");
		boolean isElementDisplayed = element.isDisplayed();
	
	but if you use chain syntax before,this change will break your chain, because it does not 	return MacacaClient anymore, if you use like this before:
	
		driver.elementById("your_element_id").click();

	you need to break this chain like this:

		driver.elementById("your_element_id");
		driver.click();

**Review by xudafeng — COMMENTED** [review]
💯 💯

**Comment by yyclaw** [comment]
snip20161228_15

这个 S 弯么？

**Comment by Yinxl** [comment]
@jsw0528 没太明白你的意思 详细点儿？

File: src/main/java/macaca/client/commands/Element.java (hunk 1 above)
Comment (Yinxl) [reply src/main/java/macaca/client/commands/Element.java:63]: 如果要找的控件不存在，elementId不会返回，这时候返回false ，在macacaClient里查找的时候通过判断false返回null对象，用户可以通过判断返回是否为null确定当前控件是否存在。

**Comment by xudafeng** [comment]
没有对齐 @Yinxl

**Comment by Yinxl** [comment]
好的 下次改一下

**Pull request merged**